static/app/stores/configStore/configProvider.tsx

```diff
@@ -1,12 +1,52 @@
-import {useReducer} from 'react';
+import {useEffect, useReducer, useRef} from 'react';
 
+import LegacyConfigStore from 'sentry/stores/configStore';
 import {ConfigContext} from 'sentry/stores/configStore/configContext';
+import {
+  configReducer,
+  makeBridgableReducer,
+} from 'sentry/stores/configStore/configReducer';
 import {Config} from 'sentry/types';
 
-import {configReducer} from './configReducer';
+interface ConfigProviderProps {
+  children: React.ReactNode;
+  initialValue: Config;
+  bridgeReflux?: boolean;
+}
+
+export function ConfigProvider(props: ConfigProviderProps) {
+  const initialBridgeRefluxValue = useRef<ConfigProviderProps['bridgeReflux']>(
+    props.bridgeReflux
+  );
+  const contextValue = useReducer(
+    makeBridgableReducer(configReducer, props.bridgeReflux ?? false),
+    props.initialValue
+  );
+
+  useEffect(() => {
+    if (initialBridgeRefluxValue.current !== props.bridgeReflux) {
+      throw new Error(
+        `bridgeReflux must not change between rerenders. This may result in undefined and out of sync behavior between the two stores. bridgeReflux changed from ${initialBridgeRefluxValue.current} -> ${props.bridgeReflux}`
+      );
+    }
+
+    if (!props.bridgeReflux) {
+      return undefined;
+    }
+
+    const [_, dispatch] = contextValue;
+
+    // @ts-ignore we need to bind the listener to a context, hence the variable name,
+    // but sadly typescript complains about shadowing here
+    const bindContext = this;
+    const unsubscribeListener = LegacyConfigStore.listen(changes => {
+      dispatch({type: 'patch', payload: changes});
+    }, bindContext);
 
-export function ConfigProvider(props: {children: React.ReactNode; initialValue: Config}) {
-  const contextValue = useReducer(configReducer, props.initialValue);
+    return () => {
+      unsubscribeListener();
+    };
+  }, [props.bridgeReflux]);
```

Comment:
I think we can probably make this simpler to not have the `timeout` issues below. Those timeouts add some risk to this since timeout'd state sync has slightly different behaviour, which might be fine for this instance, but as a pattern we might carry for a while during the refactor, might be an issue, not to mention the cognitive overhead of having 2 states that need be setup for each bridging reducer and kept in sync.

The current approach:
- **Modify ConfigStore** 
  - -> listen [context] -> dispatch [context] -> new state [context]
- **Modify context eg. 'set config value'** 
  - -> dispatch [context] -> w/ bridgeReflux ConfigStore.set -> listen[context] -> dispatch[context] (need to do setTimeout hack here)

Hypothetically we could have a stateless dispatch function that didn't use `useReducer` so it didn't perform a state update. That function could just call ConfigStore functions. Our listen could then just perform a false state update (`setUnusedState({})`) to act as a signal to context consumers. After we fully convert everything to use the new functions, then switch the dispatch function to be stateful and state to be bound to context instead.

Hypothetical change:
 - **Modify ConfigStore** 
  - -> listen [context] -> false state update [context] -> context provides singleton value of the store through the provider for now, then just a regular state later (that change is fairly low risk)
- **Modify context eg. 'set config value'** 
  - -> "dispatch (indirect function)" [context] -> calls store directly for now, then see **Modify ConfigStore**

Reply:
I think I understand what you mean - let me try and open a new PR from this branch with your approach.